File: vbnet/high-scores/HighScores.vb
Imports System
Imports System.Linq

Public Class HighScores

    Private ReadOnly _s As List(Of Integer)
    Public Sub New(scores As List(Of Integer))
        _s = scores
    End Sub

    Public Function Scores() As List(Of Integer)
        Return _s
    End Function

    Public Function Latest() As Integer
        Return _s.Last()
    End Function

    Public Function PersonalBest() As Integer
        Return _s.Max()
    End Function

    Public Function PersonalTopThree() As List(Of Integer)
        Return _s.OrderByDescending(function(s) s).Take(3).ToList()
    End Function
End Class

File: swift/high-scores/Sources/HighScores/HighScores.swift
import Foundation
struct HighScores {
    let scores: [Int]
    
    var latest: Int { scores.last ?? 0 }
    var personalBest: Int { scores.max() ?? 0 }
    var topThree: [Int] { Array(scores.sorted(by: >).prefix(3)) }
}
